src/resolve/resolved-dns-query.c

```diff
@@ -899,6 +901,12 @@ static void dns_query_accept(DnsQuery *q, DnsQueryCandidate *c) {
                         DNS_ANSWER_REPLACE(q->answer, dns_answer_ref(t->answer));
                         q->answer_rcode = t->answer_rcode;
                         q->answer_dnssec_result = t->answer_dnssec_result;
+                        q->answer_ede_rcode = t->answer_ede_rcode;
+                        if (t->answer_ede_msg) {
+                                q->answer_ede_msg = strdup(t->answer_ede_msg);
```

Comment:
could you move the strdup() before we start copying over the data from the transaction into the query object (put it into some temporary var first or so)? And if the allocation fails, then do `goto fail` (which is how we handle failures such as OOM here).

That way we either copy all fields over or none (in case of oom), which feels a lot nicer.